File: src/script/javascript/airlift/AttributeMetadata.js
var util = require('./util');

var AttributeMetadata = function(_config)
{			
	this.name = util.value(_config.name, null);
	this.type = util.value(_config.type, 'java.lang.String');
	this.isPresentable = util.value(_config.isPresentable, true);
	this.displayOrder = util.value(_config.displayOrder, 1000);	
	this.fieldSetName = util.value(_config.fieldSetName, '');
	this.label = util.value(_config.label, util.value(_config.name, ''));
	this.pluralLabel = util.value(_config.pluralLabel, '');	
	this.displayLength = util.value(_config.displayLength, 32)	;
	this.inputType = util.value(_config.inputType, 'TEXT');
	this.textAreaRows = util.value(_config.textAreaRows, 5);
	this.textAreaColumns = util.value(_config.textAreaColumns, 50);
	this.hasFormat = util.value(_config.hasFormat, '.*');
	this.readOnly = util.value(_config.readOnly, false);
	this.allowedValues = util.value(_config.allowedValues, {})
	this.dateTimePattern = util.value(_config.dateTimePattern, "MM-dd-yyyy");
	this.delimiter = util.value(_config.delimiter, ',');
	this.isPersistable = util.value(_config.isPersistable, true);
	this.maxLength = util.value(_config.maxLength, 200);
	this.minLength = util.value(_config.minLength, 0);
	this.isSearchable = util.value(_config.isSearchable, false);
	this.isIndexable = util.value(_config.isIndexable, false);
	this.isPrimaryKey = util.value(_config.isPrimaryKey, false);
	this.isUniqueKey = util.value(_config.isUniqueKey, false);
	this.concept = util.value(_config.concept, '');
	this.nullable = util.value(_config.nullable, false);
	this.immutable = util.value(_config.immutable, false);
	this.ranged = util.value(_config.ranged, false);
	this.minValue = util.value(_config.minValue, null);
	this.maxValue = util.value(_config.maxValue, null);
	this.encrypted = util.value(_config.encrypted, false);
	this.semanticType = util.value(_config.semanticType, 'NONE');
	this.mapTo = util.value(_config.mapTo, null);
};

exports.create = function(_config)
{
	return new AttributeMetadata(_config)
};

File: src/java/airlift/meta/AttributeMetadata.java
/*
 Copyright 2011, Lucid Technics, LLC.

 Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
 except in compliance with the License. You may obtain a copy of the License at

 http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in
 writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 specific language governing permissions and limitations under the License.
*/

package airlift.meta; 

public class AttributeMetadata
{
	public enum Type {  TEXT, HIDDEN, PASSWORD, TEXTAREA, CHECKBOX, RADIO, SELECT, MULTISELECT }

	public String name = "";
	public String type = "string";
	public boolean isPresentable = true;
	public int displayOrder = 1000; //By default basically use the order returned by the Java compiler.
	public String fieldSetName = "";
	public String label = "";
	public String pluralLabel = "";
	public int displayLength = 32;
	public Type inputType = Type.TEXT;
	public int textAreaRows = 5;
	public int textAreaColumns = 50;
	public String hasFormat = ".*";	
	public boolean readOnly = false;
	public String[] allowedValues = {};
	public String dateTimePattern = "MM-dd-yyyy";
	public String delimiter = ","; //to be used for attributes that are presented as checkboxes, radio buttons, and selects.

	public enum Semantic { NONE, ID, EMAIL, ADDRESS, CREDITCARD, ZIPCODE, PHONENUMBER, VERYLONGTEXT }

	public boolean isPersistable = true;
	public int maxLength = 200;
	public int minLength = 0;
	public boolean isSearchable = false;
	public boolean isIndexable = false;
	public boolean isPrimaryKey = false;
	public boolean isUniqueKey = false;
	public String concept = "";
	public boolean nullable = false;
	public boolean immutable = false;
	public boolean rangeable = false;
	public String minimumValue = null;
	public String maximumValue = null;
	public boolean encrypted = false;
	public Semantic semanticType = Semantic.NONE;

	public AttributeMetadata(String _name)
	{
		this.name = _name;
		this.label = _name;
	}
}